src/sdk/python/rtdip_sdk/pipelines/monitoring/spark/data_quality/check_value_ranges.py

```diff
@@ -0,0 +1,173 @@
+        self.logger = logging.getLogger(self.__class__.__name__)
+        if not self.logger.handlers:
+            # Prevent adding multiple handlers in interactive environments
+            handler = logging.StreamHandler()
+            formatter = logging.Formatter(
+                "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+            )
+            handler.setFormatter(formatter)
+            self.logger.addHandler(handler)
+            self.logger.setLevel(logging.INFO)
+
+    @staticmethod
+    def system_type():
+        """
+        Attributes:
+            SystemType (Environment): Requires PYSPARK
+        """
+        return SystemType.PYSPARK
+
+    @staticmethod
+    def libraries():
+        libraries = Libraries()
+        return libraries
+
+    @staticmethod
+    def settings() -> dict:
+        return {}
+
+    def check(self) -> PySparkDataFrame:
+        """
+        Executes the value range checking logic. Identifies and logs any rows where specified
+        columns exceed their defined value ranges.
+
+        Returns:
+            pyspark.sql.DataFrame:
+                Returns the original PySpark DataFrame without changes.
+        """
+        self._validate_inputs()
+        df = self.df
+
+        # For each column and its range, check for out-of-range values
+        for column, range_dict in self.columns_ranges.items():
+            min_value = range_dict.get("min", None)
+            max_value = range_dict.get("max", None)
+            inclusive = range_dict.get("inclusive", self.default_inclusive).lower()
+
+            conditions = []
+
+            # Build minimum value condition
+            if min_value is not None:
+                if inclusive in ["both", "left"]:
+                    min_condition = col(column) < min_value
+                elif inclusive in ["neither", "right"]:
+                    min_condition = col(column) <= min_value
+                else:
+                    raise ValueError(
+                        "Invalid value for 'inclusive' parameter. Must be 'both', 'neither', 'left', or 'right'."
+                    )
+                conditions.append(min_condition)
+
+            # Build maximum value condition
+            if max_value is not None:
+                if inclusive in ["both", "right"]:
+                    max_condition = col(column) > max_value
+                elif inclusive in ["neither", "left"]:
+                    max_condition = col(column) >= max_value
+                else:
+                    raise ValueError(
+                        "Invalid value for 'inclusive' parameter. Must be 'both', 'neither', 'left', or 'right'."
+                    )
+                conditions.append(max_condition)
+
+            if not conditions:
+                # Should not happen, as at least 'min' or 'max' must be provided
+                continue
+
+            # Combine all conditions with a logical OR
+            condition = reduce(or_, conditions)
```

Comment:
Shouldn't this be an `and`?

Reply:
thanks for reading the code so carefully!
i want to check if its out of bound left OR right. if i would check if a 90 is smaller than 100 AND bigger than 150, it wouldnt work

Reply:
Hmm, I see, I guess I missunderstood the code. I thought the actual conditions where being ored, not their opposite. And I thought, that if every condition has to hold, an and would be required.